packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/customization/implementation/javaparsercustomization/JavaParserClassCustomization.java

```diff
@@ -0,0 +1,139 @@
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public MethodCustomization getMethod(String methodNameOrSignature) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public ConstructorCustomization getConstructor(String constructorNameOrSignature) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public PropertyCustomization getProperty(String propertyName) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public ConstantCustomization getConstant(String constantName) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public JavadocCustomization<?> getJavadoc() {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public ConstructorCustomization addConstructor(String constructor) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public ConstructorCustomization addConstructor(String constructor, List<String> importsToAdd) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public MethodCustomization addMethod(String method) {
+        return addMethod(method, null);
+    }
+
+    @Override
+    public MethodCustomization addMethod(String method, List<String> importsToAdd) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public JavaParserClassCustomization removeMethod(String methodNameOrSignature) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public JavaParserClassCustomization rename(String newName) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public JavaParserClassCustomization setModifier(int modifiers) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public JavaParserClassCustomization addAnnotation(String annotation) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public JavaParserClassCustomization removeAnnotation(String annotation) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public JavaParserClassCustomization renameEnumMember(String enumMemberName, String newName) {
+        throw new UnsupportedOperationException(UNSUPPORTED_MESSAGE);
+    }
+
+    @Override
+    public JavaParserClassCustomization customizeAst(Consumer<CompilationUnit> astCustomization) {
+        CompilationUnit astToEdit = StaticJavaParser.parse(editor.getFileContent(fileName));
+        astCustomization.accept(astToEdit);
+        editor.replaceFile(fileName, astToEdit.toString());
+
+        return this;
```

Comment:
Confirming my understanding - so plan is, for any SDK requiring customization will "only" use `customizeAst` for all customizations. This means there will not be separate methods for individual customizations, such as `renameEnumMember`. Instead, users will use APIs from JavaParser within `customizeAst` to achieve the same results. Is this the new design

Reply:
At least in the initial migration that is the plan. In the future a more fluent design as we have with Eclipse language server could be added.